runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -220,6 +217,10 @@ private void processCheckpoint(StreamAddressSpace streamAddressSpace, Function<I
                 checkpointResolved = processCheckpointBatchByEntry(batch, filter, queue);
             }
         }
+
+        // Select correct checkpoint - Highest
+        List<Long> checkpointEntries = selectCheckpoint(getCurrentContext());
```

Comment:
what is this supposed to do ? Is it supposed to provide a list of just the highest checkpoint ?